src/index.ts

```diff
@@ -133,15 +138,28 @@ const hasValidWeekdays = (weekdays: string, alias?: boolean, allowBlankDay?: boo
     return false
   }
 
-  if (alias) {
-    const remappedWeekdays = weekdays.toLowerCase().replace(/[a-z]{3}/g, (match: string): string => {
-      return weekdaysAlias[match] === undefined ? match : weekdaysAlias[match]
-    })
-    // If any invalid alias was used, it won't pass the other checks as there will be non-numeric values in the weekdays
-    return validateForRange(remappedWeekdays, 0, allowSevenAsSunday ? 7 : 6)
+  const remappedWeekdays = alias
+    ? weekdays.toLowerCase().replace(/[a-z]{3}/g, (match: string): string => {
+        return weekdaysAlias[match] === undefined
+          ? match
+          : weekdaysAlias[match];
+      })
+    : weekdays;
+  const maxWeekdayNum = allowSevenAsSunday ? 7 : 6;
+  
+  if (allowNthWeekdayOfMonth) {
+    // see https://github.com/Airfooox/cron-validate/blob/b95aae1f3a44ad89dbfc7d1a7fca63f3b697aa14/src/helper.ts#L139
+    // and https://www.quartz-scheduler.org/documentation/quartz-2.2.2/tutorials/crontrigger.html#special-characters
+    const [weekday, occurrence, ...leftOvers] = remappedWeekdays.split('#')
+    if (leftOvers.length !== 0) {
+      return false;
+    }
+
+    return isInRange(safeParseInt(occurrence), 1, 5) &&
+      isInRange(safeParseInt(weekday), 0, maxWeekdayNum);
   }
```

Comment:
I think there is one problem here. If we enable the option (`allowNthWeekdayOfMonth: true`), it means range no longer work, even if we never use the nth weekday symbol (`#`). I assume it's the desired behaviour when the cron expression to validate contains a `#` but it should accept ranges like usual when it's not used. You can easily test it with this: 

```typescript
   it("should still allow range when nth weekday is enabled but not used", () => {
      const valid = isValidCron("* * * * mon-fri", { allowNthWeekdayOfMonth: true, alias: true });
      expect(valid).toBeTruthy(); // This fails

      const valid = isValidCron("* * * * mon-fri", { alias: true });
      expect(valid).toBeTruthy(); // This succeeds
    });
```

Reply:
Oh, silly me! Of course you're right, forgot to cover that one.

I'm thinking of another PR to automate testing combinations of seemingly unrelated flags like this one, but for now covered it manually with a one more test case.